salt/utils/etcd_util.py

```diff
@@ -202,25 +404,32 @@ def get(self, key, recurse=False):
 
         return self.tree(key)
 
-    def read(self, key, recursive=False, wait=False, timeout=None, waitIndex=None):
-        """
-        Read a value of a key.
+    def read(
+        self, key, recurse=False, wait=False, timeout=None, waitIndex=None, **kwargs
+    ):
+        recursive = kwargs.get("recursive", None)
+        if recursive is not None:
+            salt.utils.versions.warn_until(
+                "Argon",
+                "Starting with the Argon release, the recursive kwarg will be deprecated."
+                "Please use recurse instead.",
+            )
+            recurse = recursive
+        if kwargs:
+            log.warning("Invalid kwargs passed in will not be used: %s", kwargs)
```

Comment:
**suggestion** if checking kwargs here then on line 410 that should be `kwargs.pop('recursive', None)`